tools/e2e-commons/bin/wait-for-atomic-site.cjs

```diff
@@ -0,0 +1,114 @@
+const path = require( 'path' );
+const fetch = require( 'node-fetch' );
+
+// Below call should be BEFORE requiring config, so library wil pick it up.
+process.env.NODE_CONFIG_DIR = path.resolve( __dirname, '../config' );
+const { resolveSiteUrl, getSiteCredentials } = require( '../helpers/utils-helper.cjs' );
+
+function getAuthHeader() {
+	const credentials = getSiteCredentials();
+	return (
+		'Basic ' +
+		Buffer.from( credentials.username + ':' + credentials.apiPassword ).toString( 'base64' )
+	);
+}
+
+async function getJetpackVersionFromSite() {
+	let response;
+	try {
+		response = await fetch( resolveSiteUrl() + '/index.php?rest_route=/wp/v2/plugins', {
+			headers: { Authorization: getAuthHeader() },
+		} );
+
+		const contentType = response.headers.get( 'content-type' );
+		if ( contentType && ! contentType.startsWith( 'application/json' ) ) {
+			const out = await response.text();
+			if ( out.includes( 'Briefly unavailable for scheduled maintenance' ) ) {
+				console.log( 'Site is down for maintenance' );
+			}
+			return {};
+		}
+		const plugins = await response.json();
+
+		const jetpackDev = plugins.find( p => p.plugin === 'jetpack-dev/jetpack' );
+		// console.log(jetpackDev);
+		return jetpackDev.version;
+	} catch ( error ) {
+		console.error( `Logging environment details failed! ${ error }` );
+		return '';
+	}
+}
+
+async function forcePluginUpdates() {
+	const response = await fetch(
+		resolveSiteUrl() + '/index.php?rest_route=/jp-e2e/v1/beta-autoupdate',
+		{
+			method: 'POST',
+			headers: { Authorization: getAuthHeader() },
+		}
+	);
+	console.log( await response.json() );
+}
+
+async function getLatestVersion() {
+	const type = getVersionType();
+	const response = await fetch( 'https://betadownload.jetpack.me/jetpack-branches.json' );
+	const manifest = await response.json();
+
+	if ( type === 'rc' || type === 'trunk' || type === 'master' ) {
```

Comment:
I couldn't get it to work with trunk, and I think that's because `getVersionType()` returns trunk, but there is no `manifest.trunk`, jetpack-branches.json  expects 'master' in this case.